### ISSUE TITLE
Certain tool abilities reset tool damage

### ISSUE THREAD

**Issue by Pessimismus** [HbmMods/Hbm-s-Nuclear-Tech-GIT#563]
All damage a tool recieves when on of the following abilities is activated will be deleted when switching to another ability or turning the ability off:

- Silk Touch
- Fortune
- Auto-Smelter
- Auto-Shredder

When no ability is activated or one that is not listed the tool will count and save its damage like normal.

Version: 1.0.27_X4193
Only NEI and NTM installed.

**Comment by HbmMods** [comment]
A similar issue has been reported on some servers but I could never replicate it myself. Perhaps it's the server software that resets durability when things get enchanted? I know Thermos had a similar issue where using enchantment books on certain items would just not work.

**Comment by HbmMods** [comment]
Ok I think I managed to find the issue, the tool does not actually receive damage when used in all these modes, although the client still correctly (or rather, incorrectly) keeps track of the damage the tool would have gotten. The moment you switch the mode, the damage info is sent from the sever (where it hasn't taken damage) to the client, seemingly repairing the item. I've added some manual item damage calls to those abilities so the numbers add up properly.
